bot.go

```diff
@@ -33,10 +34,31 @@ type BotAPI struct {
 // NewBotAPI creates a new BotAPI instance.
 //
 // It requires a token, provided by @BotFather on Telegram.
-func NewBotAPI(token string) (*BotAPI, error) {
+func NewBotAPI(token string, proxyparam ...interface{}) (*BotAPI, error) {
```

Comment:
instead of passing `...interface{}` here, i would suggest you create a config struct that contains the data that needs to be passed.